feature/bgp/gracefulrestart/otg_tests/bgp_graceful_restart_test/bgp_graceful_restart_test.go

```diff
@@ -0,0 +1,977 @@
+	d := &oc.Root{}
+	rp := d.GetOrCreateRoutingPolicy()
+	pd := rp.GetOrCreatePolicyDefinition(name)
+	st, err := pd.AppendNewStatement("id-1")
+	if err != nil {
+		t.Fatal(err)
+	}
+	st.GetOrCreateActions().PolicyResult = pr
+	gnmi.Replace(t, dut, gnmi.OC().RoutingPolicy().Config(), rp)
+}
+
+// configureDUT configures all the interfaces and network instance on the DUT.
+func configureDUT(t *testing.T, dut *ondatra.DUTDevice) {
+	dc := gnmi.OC()
+	i1 := dutSrc.NewOCInterface(dut.Port(t, "port1").Name(), dut)
+	gnmi.Replace(t, dut, dc.Interface(i1.GetName()).Config(), i1)
+
+	i2 := dutDst.NewOCInterface(dut.Port(t, "port2").Name(), dut)
+	gnmi.Replace(t, dut, dc.Interface(i2.GetName()).Config(), i2)
+
+	t.Log("Configure/update Network Instance")
+	fptest.ConfigureDefaultNetworkInstance(t, dut)
+
+	if deviations.ExplicitPortSpeed(dut) {
+		fptest.SetPortSpeed(t, dut.Port(t, "port1"))
+		fptest.SetPortSpeed(t, dut.Port(t, "port2"))
+	}
+	if deviations.ExplicitInterfaceInDefaultVRF(dut) {
+		fptest.AssignToNetworkInstance(t, dut, i1.GetName(), deviations.DefaultNetworkInstance(dut), 0)
+		fptest.AssignToNetworkInstance(t, dut, i2.GetName(), deviations.DefaultNetworkInstance(dut), 0)
+	}
+}
+
+func verifyPortsUp(t *testing.T, dev *ondatra.Device) {
+	t.Helper()
+	for _, p := range dev.Ports() {
+		status := gnmi.Get(t, dev, gnmi.OC().Interface(p.Name()).OperStatus().State())
+		if want := oc.Interface_OperStatus_UP; status != want {
+			t.Errorf("%s Status: got %v, want %v", p, status, want)
+		}
+	}
+}
+
+type bgpNeighbor struct {
+	as         uint32
+	neighborip string
+	isV4       bool
+}
+
+func buildNbrList() []*bgpNeighbor {
+	nbr1v4 := &bgpNeighbor{as: ateAS, neighborip: ateSrc.IPv4, isV4: true}
+	nbr1v6 := &bgpNeighbor{as: ateAS, neighborip: ateSrc.IPv6, isV4: false}
+	nbr2v4 := &bgpNeighbor{as: dutAS, neighborip: ateDst.IPv4, isV4: true}
+	nbr2v6 := &bgpNeighbor{as: dutAS, neighborip: ateDst.IPv6, isV4: false}
+	return []*bgpNeighbor{nbr1v4, nbr2v4, nbr1v6, nbr2v6}
+}
+
+func bgpWithNbr(as uint32, keepaliveTimer uint16, nbrs []*bgpNeighbor, dut *ondatra.DUTDevice) *oc.NetworkInstance_Protocol {
+	d := &oc.Root{}
+	ni1 := d.GetOrCreateNetworkInstance(deviations.DefaultNetworkInstance(dut))
+	niProto := ni1.GetOrCreateProtocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "BGP")
+	bgp := niProto.GetOrCreateBgp()
+
+	g := bgp.GetOrCreateGlobal()
+	g.As = ygot.Uint32(as)
+	g.GetOrCreateAfiSafi(oc.BgpTypes_AFI_SAFI_TYPE_IPV4_UNICAST).Enabled = ygot.Bool(true)
+	g.GetOrCreateAfiSafi(oc.BgpTypes_AFI_SAFI_TYPE_IPV6_UNICAST).Enabled = ygot.Bool(true)
+	g.RouterId = ygot.String(dutDst.IPv4)
+	bgpgr := g.GetOrCreateGracefulRestart()
+	bgpgr.Enabled = ygot.Bool(true)
+
+	pg := bgp.GetOrCreatePeerGroup(peerv4GrpName)
+	pgGrV4 := pg.GetOrCreateGracefulRestart()
+	pgGrV4.Enabled = ygot.Bool(true)
+	pgGrV4.RestartTime = ygot.Uint16(grRestartTime)
+	pgGrV4.StaleRoutesTime = ygot.Uint16(grStaleRouteTime)
```

Comment:
These (RestartTime and StaleRoutesTime) are indicated in the README as set at the 'global' level. (I believe the existing ATE test does do that.) Might also imply changes in `checkBgpGRConfig`?